### PR TITLE
[0.10] Bundle User & System Preferences

### PR DESCRIPTION
**Pull request by v1r0x**
@Severino Please have a look at this PR

### CODE REVIEW

File: resources/js/components/Preferences.vue [hunk 1]
```diff
@@ -173,81 +232,196 @@
...
+                                preferences: subcategory.preferences.slice(),
+                            };
+                            if(subcategory.custom) {
+                                cats.system[k].custom = true;
+                                cats.system[k].title = subcategory.title;
+                            }
+                        } else {
+                            cats.system[k].preferences = cats.system[k].preferences.concat(subcategory.preferences.slice());
+                        }
+                    }
```
Comment (Severino) [inline resources/js/components/Preferences.vue]: This is 2 times the same 2 code blocks (DRY). I refactored this and will push the changes soon.

Comment (Severino) [reply resources/js/components/Preferences.vue]: Also I renamed cats to categories, as I was confused what cats 🐱 are doing in our application

File: app/Plugin.php [hunk 2]
```diff
@@ -205,9 +202,15 @@ public function handleRemove() {
...
             $this->handleUninstall();
+            $this->rollbackMigrations();
+            $this->removePermissions();
+            $this->uninstallPresets();
         }
 
+        $this->removePreferences();
         sp_remove_dir(base_path("app/Plugins/$this->name"));
+
+        $this->delete();
```
Comment (Severino) [inline app/Plugin.php:213]: Great, now the migrations will be only rolled back if the plugin os permanently removed, right?

File: resources/js/components/Preferences.vue (hunk 1 above)
Comment (Severino) [reply resources/js/components/Preferences.vue]: refactored in #[b638606](https://github.com/DH-Center-Tuebingen/Spacialist/pull/494/commits/b638606b4bf415c62f9561ab2f8854cffd68df87)

File: app/Http/Controllers/HomeController.php [hunk 3]
```diff
@@ -38,7 +38,10 @@ public function getGlobalData() {
             $locale = \App::getLocale();
         }
 
-        $sysPrefs = Preference::getPreferences();
+        $sysPrefsValues = [];
+        foreach(Preference::getPreferences() as $k => $p) {
+            $sysPrefsValues[$k] = $p['value'];
+        }
```
[the remark this answers is not in the document]
Comment (v1r0x) [reply app/Http/Controllers/HomeController.php]: You are right. I simplified it and also added `$simple` as param to `getUserPreferences()`

File: resources/js/components/Preferences.vue (hunk 1 above)
Comment (v1r0x) [reply resources/js/components/Preferences.vue]: I like 🐱
:sweat_smile:

File: app/Plugin.php (hunk 2 above)
Comment (v1r0x) [reply app/Plugin.php:213]: Yes. I'm not sure if we should still remove permissions and presets. We'll see if this leads to unexpected behaviour with inactive plugins

**Comment by v1r0x** [comment]
@Severino I merged `release-0.10` into this branch. No problems for me. Anything from your side? Or can we merge?

**Comment by Severino** [comment]
Fine for me! :)

**Pull request merged**